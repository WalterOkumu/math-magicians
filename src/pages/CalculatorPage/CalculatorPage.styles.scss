.calculator-page-container {
  display: flex;
  flex-direction: row;
  padding: 20px;

  .text {
    width: 45%;

    h4 {
      font-size: 25px;
    }
  }

  .calculator {
    width: 55%;
  }
}
