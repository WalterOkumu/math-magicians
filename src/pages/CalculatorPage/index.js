import React from 'react';
import Calculator from '../../components/Calculator';
import './CalculatorPage.styles.scss';

const CalculatorPage = () => (
  <div className="calculator-page-container">
    <div className="text">
      <h4>
        Let&#39;s do some math!
      </h4>
    </div>
    <Calculator className="calculator" />
  </div>
);

export default CalculatorPage;
