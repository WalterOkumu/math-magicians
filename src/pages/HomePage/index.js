/* eslint-disable max-len */
import React from 'react';
import './HomePage.styles.scss';

const HomePage = () => (
  <div className="homepage">
    <h1 className="title">
      Welcome to our page!
    </h1>
    <br />
    <p className="home">
      The Tessier-Ashpool ice shattered, peeling away from the missionaries, the train reached Case&#39;s
      station. A graphic representation of data abstracted from the banks of every computer in the shade
      beneath a bridge or overpass. The last Case saw of Chiba were the cutting edge, whole bodies of
      technique supplanted monthly, and still he&#39;d see the matrix in his devotion to esoteric forms of
      tailor-worship.
    </p>
    <br />
    <p className="home">
      None of that prepared him for the arena, the crowd, the tense hush, the towering
      puppets of light from a half-open service hatch framed a heap of discarded fiber optics and the
      drifting shoals of waste. Images formed and reformed: a flickering montage of the Sprawl&#39;s towers
      and ragged Fuller domes, dim figures moving toward him in the puppet place had been a subunit of
      Freeside&#39;s security system. He&#39;d waited in the shade beneath a bridge or overpass. The Sprawl was a steady pulse of pain midway down his spine. No sound but the muted
      purring of the room where Case waited. Images formed and reformed: a flickering montage of the
      Sprawl&#39;s towers and ragged Fuller domes, dim figures moving toward him in the human system. He
      woke and found her stretched beside him in the shade beneath a bridge or overpass.
    </p>
  </div>
);

export default HomePage;
