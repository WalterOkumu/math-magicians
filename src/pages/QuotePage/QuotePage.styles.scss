.container {
  margin: auto;
  width: 50%;
  padding: 50px;
  text-align: center;

  .title {
    font-size: 40px;
    padding: 10px;
  }

  p {
    font-size: 20px;
    font-weight: bold;
  }

  span {
    font-style: italic;
    font-size: 15px;
  }
}
