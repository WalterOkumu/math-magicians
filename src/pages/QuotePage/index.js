import React from 'react';
import './QuotePage.styles.scss';
import quotes from '../../data/quotes.json';

const QuotePage = () => {
  const getRandomIndex = () => {
    const min = 0;
    const max = 4;

    return Math.floor(Math.random() * (max - min + 1)) + min;
  };

  const quoteObj = quotes[getRandomIndex()];

  return (
    <div className="container">
      <h2 className="title">Math Quote</h2>
      <p>
        {quoteObj.quote}
      </p>
      <span>
        -
        {quoteObj.author}
      </span>
    </div>
  );
};

export default QuotePage;
