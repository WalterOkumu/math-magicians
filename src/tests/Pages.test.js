import { render, screen } from '@testing-library/react';
import HomePage from '../pages/HomePage';
import CalculatorPage from '../pages/CalculatorPage';
import QuotePage from '../pages/QuotePage';

describe('Test React Components', () => {
  test('Test Home Page is rendered', () => {
    render(<HomePage />);

    expect(screen.getAllByText('Welcome to our page!')).toMatchSnapshot();
  });

  test('Test Calculator Page is rendered', () => {
    render(<CalculatorPage />);

    expect(screen.findAllByTitle('Let&#39;s do some math!')).toMatchSnapshot();
  });

  test('Test Quote Page is rendered', () => {
    render(<QuotePage />);

    expect(screen.findAllByTitle('Math Quote')).toMatchSnapshot();
  });
});
