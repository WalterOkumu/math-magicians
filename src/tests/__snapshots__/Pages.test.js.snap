// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test React Components Test Calculator Page is rendered 1`] = `Promise {}`;

exports[`Test React Components Test Home Page is rendered 1`] = `
Array [
  <h1
    class="title"
  >
    Welcome to our page!
  </h1>,
]
`;

exports[`Test React Components Test Quote Page is rendered 1`] = `Promise {}`;
